src/js/Manager/RecommendationManager.js

```diff
@@ -66,12 +84,22 @@ class RecommendationManager {
         let query = new SearchQuery('or');
         query
             .where(
-                query.field('host').equals(url.host),
-                query.field('url').contains(url.pathname.length > 1 ? url.host + url.pathname:url.host)
+                // search by domain
+                (this.options.searchQuery === "domain" ? query.field('host').contains(this.getSearchDomainFromHost(url.host)): 
+                    // search by hostname
+                    (this.options.searchQuery === "host" ? query.field('host').startsWith(url.host.split(':')[0]): 
+                        // search by hostname and port
+                        (this.options.searchQuery === "hostport" ? query.field('host').equals(url.host): 
+                            // search exact
+                            (this.options.searchQuery === "exact" ? query.field('url').equals(url.protocol + "//" + url.host + (url.pathname.length > 1 ? url.pathname: "")):
+                            "")
+                        )
+                    )
+                )
```

Comment:
This is really hard to read. It would probably be better to pass the query to a different function here that returns the actual filter. That function can then have ifs for all the different options.

```
        query
            .where(...this._getFilterQuery(this.options.searchQuery, query))

[...]

_getFilterQuery(mode, query) {
  if(mode === 'host') {
    return [
      query.field('host').equals(url.host)
    ]
  }
  [...]
}
```

Reply:
I moved the filter expression in a seperate function as requested.